hpx/runtime/serialization/array.hpp

```diff
@@ -99,6 +98,13 @@ namespace hpx { namespace serialization
         return array<T>(begin, size);
     }
 
+    // implement serialization for arrays
+    template <class Archive, class T, std::size_t N>
+    void serialize(Archive& ar, T (&a)[N], const unsigned int /* version */)
+    {
+        ar & hpx::serialization::make_array(a, N);
+    }
+
     // implement serialization for boost::array
```

Comment:
I'm feeling uneasy when I see this change. Does that mean that we now can pass plain arrays as arguments to HPX' API functions (like `apply` and `async`)?